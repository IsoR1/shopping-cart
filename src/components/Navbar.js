import React from "react";
import { Link } from "react-router-dom";

const Navbar = () => {
  return (
    <>
      <nav>
        <ul className="header-ul">
          <li className="header-li">
            <Link to="/shop">Shop</Link>
          </li>
          <li className="header-li">
            {/* <Link>Search</Link> */}
            Search
          </li>
          <li className="header-li">
            <Link to="/cart">Cart</Link>
          </li>
        </ul>
      </nav>
    </>
  );
};

export default Navbar;
