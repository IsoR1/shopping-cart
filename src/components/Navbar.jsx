import React from "react";
import Search from "./Search";
import { Link } from "react-router-dom";

const Navbar = ({ cartItemCount, addItemToCart, removeItemFromCart }) => {
  return (
    <>
      <nav>
        <ul className="header-ul">
          <li className="header-li">{<Search />}</li>
          <li className="header-li">
            {/* <Link>Search</Link> */}
            {/* Search */}
            <Link to="/shop">Shop</Link>
          </li>
          <li className="header-li">
            <Link
              to="/cart"
              addItemToCart={addItemToCart}
              removeItemFromCart={removeItemFromCart}
            >
              Cart <span>{cartItemCount}</span>
            </Link>
          </li>
        </ul>
      </nav>
    </>
  );
};

export default Navbar;
