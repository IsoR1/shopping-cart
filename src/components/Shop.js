import React, { useState, useEffect } from "react";
import categories from "../data/categories";
import Sidebar from "./Sidebar";
import { render } from "@testing-library/react";
import { useParams } from "react-router-dom";

// const Shop = ({ setSearchParams }) => {
const Shop = () => {
  // const filteredCategory = categories.find((el) => el[0].category === category);

  // const location = useLocation();
  const { category } = useParams();
  const filteredCategories = categories.find(
    (el) => el[0].category === category
  );
  const renderContent = () => {
    if (filteredCategories) {
      return Object.keys(filteredCategories).map((key) => (
        //   Object.keys(el).map(
        //     (key) => console.log(el[key])
        <div className="shop-card">
          <img src={filteredCategories[key].images[0]} />
          <span className="shop-card-span">{filteredCategories[key].name}</span>
        </div>
      ));
    }
    // if (cat === "hats") {
    // {
    //   categories.map((el) =>
    //     Object.keys(el).map((key) => (
    //       <div className="shop-card">
    //         <img src={el[key].images[0]} />
    //         <span className="shop-card-span">{el[key].name}</span>
    //       </div>
    //     ))
    //   );
    // }
    // console.log(type);
    // }
  };
  return (
    <div className="shop-container">
      <Sidebar />
      <div id="shop-div">
        {/* {categories.map((el) =>
          Object.keys(el).map((key) => (
            <div className="shop-card">
              <img src={el[key].images[0]} />
              <span className="shop-card-span">{el[key].name}</span>
            </div>
          )) */}
        {/* )} */}
        {renderContent()}
      </div>
    </div>
  );
};

export default Shop;
