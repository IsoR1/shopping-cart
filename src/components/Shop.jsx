import React, { useState, useEffect } from "react";
import categories from "../data/categories";
import Sidebar from "./Sidebar";
import { render } from "@testing-library/react";
import { json, useParams } from "react-router-dom";
import ShopCard from "./ShopCard";
// ADDED ALL PRODUCTS STOPPED RENDERING FROM PRODUCTS
const Shop = ({
  searchInput,
  products,
  setProducts,
  allProducts,
  setAllProducts,
  categories,
  setCategories,
  selectedCategory,
  setSelectedCategory,
}) => {
  const { category } = useParams();

  // const filteredCategories = categories.find(
  //   (el) => el[0].category === category
  // );

  // if (category) {
  //   console.log("categorys:", category);
  // }
  useEffect(() => {
    if (category) {
      setSelectedCategory(category);
      // console.log("new cat: ");
    } else if (!category) {
      setProducts(allProducts);
    }

    // test
    console.log(category, "cat");
    if (typeof category === "undefined") {
      setProducts(allProducts);
    }
  }, [category]);

  useEffect(() => {
    // categories.map((el) => {
    //   Object.keys(el).map((key) =>
    //     // <ShopCard
    //     //   props={{

    //     //   }}
    //     // />
    //     // console.log(el[key])
    //     );
    //   });
    console.log("searchInput changed:", searchInput);
  }, [searchInput]);

  useEffect(() => {
    // let filteredProducts = products;
    console.log("l");
    console.log("products: ", allProducts);
    console.log(selectedCategory);
    if (selectedCategory) {
      const filteredProducts = allProducts.filter(
        (prod) => prod.category === selectedCategory.replace(/-/g, " ")
      );

      console.log("filtered", filteredProducts);

      setProducts(filteredProducts);
    }
  }, [selectedCategory, setProducts]);

  const renderContent = () => {
    // if (selectedCategory) {
    //   console.log("yes");
    //   console.log("testing test test: ", selectedCategory);
    //   return products
    //     .filter((prod) => prod.category === selectedCategory)
    //     .map((prod) => (
    //       <ShopCard
    //         props={{
    //           key: prod.id,
    //           images: prod.image,
    //           name: prod.title,
    //           id: prod.id,
    //           price: prod.price,
    //           category: prod.category,
    //         }}
    //       />
    //     ));
    // } else {
    // console.log(prod.category);
    return products.map((prod) => (
      <ShopCard
        props={{
          key: prod.id,
          images: prod.image,
          name: prod.title,
          id: prod.id,
          price: prod.price,
          category: prod.category,
        }}
      />
    ));
    // }
  };

  return (
    <div className="shop-container" data-testid="shop-con">
      <Sidebar categories={categories} products={products} />
      <div id="shop-div" data-testid="shop-div">
        {renderContent()}
      </div>
    </div>
  );
};

export default Shop;

// const renderContent = () => {
// Render all shop items
// if (!filteredCategories) {
//   return categories.map((el) =>
//     Object.keys(el).map((key) => (
//       <ShopCard
//         props={{
//           images: el[key].images,
//           name: el[key].name,
//           id: el[key].id,
//           price: el[key].price,
//           category: el[key].category,
//         }}
//       />
//     ))
//   );
// }
// // Render search item
// if (searchInput) {
//   return categories.map((el) => {
//     Object.keys(el).map(
//       (key) =>
//         el[key].name.includes(searchInput) && (
//           <ShopCard
//             props={{
//               images: el[key].images,
//               name: el[key].name,
//               id: el[key].id,
//               price: el[key].price,
//               category: el[key].category,
//             }}
//           />
//         )
//     );
//   });
// }
// // Render category that's clicked in the sidebar
// if (filteredCategories) {
//   return Object.keys(filteredCategories).map((key) => (
//     <ShopCard
//       props={{
//         images: filteredCategories[key].images,
//         name: filteredCategories[key].name,
//         id: filteredCategories[key].id,
//         price: filteredCategories[key].price,
//         category: filteredCategories[key].category,
//       }}
//     />
//   ));
// }
// };

// OLD WAY

// useEffect(() => {
//   let url = "https://fakestoreapi.com/products";

//   // Check if a category is selected
//   if (category) {
//     url = `https://fakestoreapi.com/products/category/${category.replace(
//       /-/g,
//       " "
//     )}`;
//   }

//   fetch(url)
//     .then((res) => res.json())
//     .then((data) => {
//       const modifiedData = data.map((el) => ({
//         ...el,
//         quantity: 1,
//       }));
//       setProducts(modifiedData);
//       // console.log(modifiedData);
//     })
//     .catch((error) => {
//       console.log("Error fetching products;", error);
//     });
// }, [category]);

// useEffect(() => {
//   fetch("https://fakestoreapi.com/products")
//     .then((res) => res.json())
//     .then((data) => setProducts(data))
//     .catch((error) => {
//       console.log("Error fetching products;", error);
//     });
// }, []);

// if (category) {
// useEffect(() => {
//   fetch(
//     `https://fakestoreapi.com/products/category/${category.replace(
//       /-/g,
//       " "
//     )}`
//   )
//     .then((res) => res.json())
//     .then((data) => setProducts(data));
//   console.log("test");
// }, [category]);

// }

// const renderContent = () => {
// if (category) {
//   // products.map((prod) => {
//   //   // if (prod.category.replace(/\s+/g, "-") === category) {
//   //   // <ShopCard
//   //   //   props={{
//   //   //     images: prod.image,
//   //   //     name: prod.title,
//   //   //     id: prod.id,
//   //   //     price: prod.price,
//   //   //     category: prod.category,
//   //   //   }}
//   //   // />;
//   //   console.log(prod.category);
//   //   // }
//   // });
//   fetch(
//     `https://fakestoreapi.com/products/category/${category.replace(
//       /-/g,
//       " "
//     )}`
//   )
//     .then((res) => res.json())
//     .then((data) => setProducts(data));
//   console.log("test");
// }

// return products.map((prod) => (
//   <ShopCard
//     props={{
//       images: prod.image,
//       name: prod.title,
//       id: prod.id,
//       price: prod.price,
//       category: prod.category,
//     }}
//   />
// ));
// };
