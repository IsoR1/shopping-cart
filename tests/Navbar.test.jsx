import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Route, Routes } from "react-router-dom";
import Navbar from "../src/components/Navbar";
import { BrowserRouter as Router } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import Shop from "../src/components/Shop";

describe("Navbar", () => {
  it("renders the navbar", () => {
    const navBar = render(
      <Router>
        <Navbar />
      </Router>
    );

    expect(navBar).toMatchSnapshot();
  });

  it("redirects to shop", async () => {
    render(
      <Router>
        <Navbar />
      </Router>
    );

    const shopLink = screen.getByRole("link", { name: "Shop" });
    await userEvent.click(shopLink);
    expect(window.location.pathname).toBe("/shop");
  });
  it("redirects to cart", async () => {
    render(
      <Router>
        <Navbar />
      </Router>
    );

    const cartLink = screen.getByRole("link", { name: "Cart" });
    await userEvent.click(cartLink);
    expect(window.location.pathname).toBe("/cart");
  });
});
