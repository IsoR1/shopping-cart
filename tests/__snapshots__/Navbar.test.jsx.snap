// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > renders the navbar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <ul
          class="header-ul"
        >
          <li
            class="header-li"
          >
            <div
              class="search-wrapper"
            >
              <div
                class="search-box"
              >
                <input
                  class=""
                  placeholder="Search"
                  type="text"
                />
                <div
                  class="icon "
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-magnifying-glass i-fa"
                    data-icon="magnifying-glass"
                    data-prefix="fas"
                    data-testid="search-icon"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </li>
          <li
            class="header-li"
          >
            <a
              data-testid="shop-link"
              href="/shop"
            >
              Shop
            </a>
          </li>
          <li
            class="header-li"
          >
            <a
              href="/cart"
            >
              Cart 
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <ul
        class="header-ul"
      >
        <li
          class="header-li"
        >
          <div
            class="search-wrapper"
          >
            <div
              class="search-box"
            >
              <input
                class=""
                placeholder="Search"
                type="text"
              />
              <div
                class="icon "
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-magnifying-glass i-fa"
                  data-icon="magnifying-glass"
                  data-prefix="fas"
                  data-testid="search-icon"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </li>
        <li
          class="header-li"
        >
          <a
            data-testid="shop-link"
            href="/shop"
          >
            Shop
          </a>
        </li>
        <li
          class="header-li"
        >
          <a
            href="/cart"
          >
            Cart 
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
